content/switching-to-v0-mangling-on-nightly.md

```diff
@@ -0,0 +1,171 @@
+  compatibility with tools on varied platforms
+- It tries to stay efficient and avoid unnecessarily long names and
+  computationally-expensive decoding 
+
+However, rustc is not the only tool that interacts with Rust symbol names: the
+aforementioned debuggers, profilers and other tools all need to be updated to
+understand Rust's v0 symbol mangling scheme so that Rust's users can continue
+to work with Rust binaries using all the tools they're used to. Furthermore, all
+of those tools need to have new releases cut and then those releases need to be
+picked up by distros. This takes time!
+
+Fortunately, the compiler team now believe that support for our v0 mangling
+scheme is now sufficiently widespread that it can start to be used by default by
+rustc.
+
+#### Benefits
+
+Reading Rust backtraces, or using Rust with debuggers, profilers and other
+tools that operate on compiled Rust code, will be able to output much more
+useful and readable names. This will especially help with async code,
+closures and generic functions.
+
+It's easy to see the new mangling scheme in action, consider the following
+example:
+
+```rust
+fn foo<T>() {
+    panic!()
+}
+
+fn main() {
+    foo::<Vec<(String, &[u8; 123])>>();
+}
+```
+
+With the legacy mangling scheme, all of the useful information about the generic
+instantiation of `foo` is lost in the symbol `f:foo`..
+
+```
+thread 'main' panicked at f.rs:2:5:
+explicit panic
+stack backtrace:
+  0: std::panicking::begin_panic
+    at /rustc/d6c...582/library/std/src/panicking.rs:769:5
+  1: f::foo
+  2: f::main
+  3: core::ops::function::FnOnce::call_once
+note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
+```
+
+..but with the v0 mangling scheme, the useful details of the generic instantiation
+are preserved with  `f::foo::<alloc::vec::Vec<(alloc::string::String, &[u8; 123])>>`:
+
+```
+thread 'main' panicked at f.rs:2:5:
+explicit panic
+stack backtrace:
+  0: std::panicking::begin_panic
+    at /rustc/d6c...582/library/std/src/panicking.rs:769:5
+  1: f::foo::<alloc::vec::Vec<(alloc::string::String, &[u8; 123])>>
+  2: f::main
+  3: <fn() as core::ops::function::FnOnce<()>>::call_once
+note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
+```
+
+#### Possible drawbacks
+
+Symbols using the v0 mangling scheme can be larger than symbols with the
+legacy mangling scheme, which can result in a slight increase in linking
+times. Fortunately this impact should be minor, especially with modern
+linkers like lld, which Rust [will now default to on some targets][switch-to-lld].
+
+Some old versions of tools/distros or niche tools that the compiler team are
+unaware of may not have had support for the v0 mangling scheme added.
+
+In any case, using the new mangling scheme can be disabled if any problem
+occurs: use the `-Csymbol-mangling-version=legacy -Zunstable-options` flag
+to revert to using the legacy mangling scheme.
+
+#### Summary
```

Comment:
I think we should have a section for tool owners that want to add support pointing at rustc-demangle which has a C and Rust implementation of the v0 demangler, which hopefully makes it easier for people to quickly add support if they find it's missing.

(I can try to write some text later, don't have time right this moment).